apps/storybook/src/stories/identity/Address.tsx

```diff
@@ -1,34 +1,62 @@
-import { getAddress, Hex } from 'viem';
-import { useEnsAddress } from 'wagmi';
-import { normalize } from 'viem/ens'
-import { getShortAddress } from '../../utils/address';
-import { useMemo } from 'react';
+import {
+	Tooltip,
+	TooltipContent,
+	TooltipProvider,
+	TooltipTrigger,
+} from "@/components/ui/tooltip";
+import { Check, Copy } from "lucide-react";
+import { useMemo, useState } from "react";
+import { Hex, getAddress } from "viem";
+import { normalize } from "viem/ens";
+import { useEnsAddress } from "wagmi";
+import { getShortAddress } from "../../utils/address";
 
-
-export const Address = ({ address, isShort = true }: {
-    address: Hex,
-    isShort?: boolean
+export const Address = ({
+	address,
+	isShort = true,
+}: {
+	address: Hex;
+	isShort?: boolean;
 }) => {
-    const addressDisplayed = useMemo(() => {
-        return isShort ? getShortAddress(address) : getAddress(address);
-    }, [address, isShort])
+	const [copied, setCopied] = useState(false);
 
-    return (
-        <div>
-            {addressDisplayed}
-        </div>
-    )
-}
+	const copyToClipboard = () => {
+		navigator.clipboard.writeText(getAddress(address));
+		setCopied(true);
+		setTimeout(() => setCopied(false), 2000);
+	};
 
-export const AddressFromName = ({ name }: {
-    name: string
-}) => {
+	const addressDisplayed = useMemo(() => {
+		return isShort ? getShortAddress(address) : getAddress(address);
```

Comment:
nit: It feels slightly unnatural that `isShort={false}` is used to display the full address. I don’t mind having the short form as the default, but it might feel more intuitive to switch to the full address with something like `isLong={true}` or `isFull={true}`.

Reply:
hmm, thats right, people are mostly looking to display the short one by default, will make the changes